src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -216,17 +218,35 @@ def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper
     return task_list_file, len(genome_ids_chunks)
 
 
-def _create_batch_script(job_dir, task_list_file, n_jobs):
+def _cal_node_num(tool, n_jobs):
+    """
+    Calculate the number of nodes required for the task
+    """
+
+    tool_exe_time = TASK_META[tool]['exe_time']
+    jobs_per_node = NODE_TIME_LIMIT * 60 // tool_exe_time
+
+    num_nodes = min(math.ceil(n_jobs / jobs_per_node), MAX_NODE_NUM)
+
+    return num_nodes
+
+
+def _create_batch_script(job_dir, task_list_file, n_jobs, tool):
     """
     Create the batch script (submit_taskfarmer.sl) for job submission
     """
+
+    node_num = _cal_node_num(tool, n_jobs)
+
     batch_script = f'''#!/bin/sh
 
-#SBATCH -N {n_jobs + 1} -c 64
+#SBATCH -N {node_num + 1} -c 256
 #SBATCH -q regular
-#SBATCH --time=4:00:00
+#SBATCH --time={NODE_TIME_LIMIT}:00:00
```

Comment:
If there are so few jobs you only need one node for less time it'd be nice to request that amount of time, but that can be done later IMO

Reply:
👍 nice things to have.

Reply:
:+1: